gnd/src/main/java/com/google/android/gnd/ui/common/photoview/PhotoView.java

```diff
@@ -0,0 +1,58 @@
+package com.google.android.gnd.ui.common.photoview;
+
+import android.content.Context;
+import android.net.Uri;
+import android.util.AttributeSet;
+import android.view.LayoutInflater;
+import androidx.fragment.app.Fragment;
+import com.google.android.gnd.databinding.PhotoViewBinding;
+import com.google.android.gnd.model.observation.Response;
+import com.google.android.gnd.system.StorageManager;
+import com.google.android.material.card.MaterialCardView;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+public class PhotoView extends MaterialCardView {
```

Comment:
It's probably my fault for being vague in previous comments, but we're generally trying to avoid creating custom Views unless absolutely necessary, since they're hard to test and easy to get wrong (as you discovered)! Instead, could this view be reduced to a layout xml that can be included in the photo_input_field layout, which in turn gets added by the Fragment?

Reply:
Thanks for the idea. It all looks better than the first approach I took.